externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/TrinoStatusPrinter.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.trino
+
+import java.util.concurrent.TimeUnit._
+
+import io.airlift.units.DataSize
+import io.airlift.units.Duration
+import io.trino.client.StatementClient
+
+import org.apache.kyuubi.engine.trino.ProgressFormatUtils._
+import org.apache.kyuubi.operation.log.OperationLog
+
+/**
+ * Copy from trino-cli
+ *   1. use OperationLog instead PrintStream to out status info
+ */
+object TrinoStatusPrinter {
+
+  def printFinalInfo(
+      client: StatementClient,
+      operationLog: OperationLog,
+      debug: Boolean = false): Unit = {
+    val out = new TrinoConsoleProgressBar(operationLog)
+    val results = client.finalStatusInfo()
+    val stats = results.getStats()
+
+    val wallTime = Duration.succinctDuration(stats.getElapsedTimeMillis(), MILLISECONDS)
```

Comment:
why not use millis directly？

Reply:
> why not use millis directly？

It is used to covert millis to Duration, to pass to functions in `FormatUtils`.

Reply:
what is FormatUtils？

Reply:
> what is FormatUtils？

https://github.com/apache/incubator-kyuubi/pull/2896/files#diff-16f6ae2d81713d0c45408cbd4a43eda8612a42e461691f04033672ad858a5e8dR34-R37
```
/** Copy from trino-cli `FormatUtils` */
public final class ProgressFormatUtils {
  private ProgressFormatUtils() {}
```
Used to format output data and process ba.